### DISCUSSION TITLE
Release 15.18.0 — Radio Button

### DISCUSSION THREAD

**Discussion by ipozuelo** [Telefonica/mistica-design#1923] [Proposals]
Buenas, queremos un comportamiento que no logramos hacer con los componentes de mistica. Utilizamos RadioGroup y RadioButton para seleccionar dos opciones. Pero en un caso, una de las opciones tiene que estar deshabilitada dependiendo de la información que nos llega. Solo vemos la opción de deshabilitar en el RadioGroup, queremos en el RadioButton. Hemos visto que en RadioButton hay un render que tiene algo de disabled, pero no logramos hacerlo funcionar. Está dentro de una BoxedRow, que si está deshabilitada, pero se puede seguir pulsando el RadioButton. 
![image](https://github.com/user-attachments/assets/052d76ad-6904-4279-b4d5-bfd13d15158b)

**Comment by ipozuelo** [comment]
![image](https://github.com/user-attachments/assets/41011b8d-63e9-4969-9043-e96017192a4a)
esto es lo que queremos evitar

**Comment by yceballost** [comment]
Hola @ipozuelo ! Lo analizamos!

Puedes pasarnos un ejemplo de lo que tienes construido en Playroom?

[mistica-web.vercel.app/playroom](mistica-web.vercel.app/playroom)

**Comment by yceballost** [comment]
A día de hoy no es posible tener un valor de un radio group disabled. O están todos o ninguno.

Me parece que puede tener sentido lo que comentas, voy a mover esto a discussion para hablarlo con el equipo y de ahí decidir si se incluiría o no.

**Comment by yceballost** [comment]
Buenas, perdona la demora sobre esto.

Hemos visto varias cosas:
- No deberías usar un radio con el extra, sino usar la prop radioValue que ya incorpora el componente Row ([ejemplo](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIAEAeASgQygSwgcQCcIBXAByVgDMNiAbAFwDUNbiYBeAHRACMMA7ARm4A%2BTvySSpk9BADuAGWwBnemInTNsueM16kGZcpj12wFAElI-AMoALCKTQwA5nQwEkAehEBfXfrS9Nj0tBzcAEJCgtwBgZKwymAE2KTBEPxcIACaMLS08rEa8UgEWLjMrOG80cIgcfo%2BDXrazfqGxqbmVhn2js5utB7efm16waHVAIKkpGFFJQkwSSlpuJnchDAw-AuLZTgQlWxZGLPz9cWN6oEoXmjySqo3Wvfl%2BERkIiAANCD0dhgAFtlggANogACyEAAbip6B4QABdP5ybBQAHKcEAZgAbAAGJG%2BIA))
- Desde Mística debemos solucionar que cuando una lista esté disabled también se inactive el radioValue (Evitar esto → [Ejemplo](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AlAhlAlhA4gJwgFcAHAAlgDM1CAbAFwDU0bCYBeAHRACM0A7fmi4A%2BDn1KlUEAO4AZTAGc6o8RMkoZYtWrQKFMOm2BIAkpD4BlABYRiKGAHNaaPKQD0wgL5btpOpjoadi4AIUEBLh9tWAUwPExifwg%2BThAATRgaGk0QKLU8DGwmFmCecKFc1QkPPKlpPIksBTRuIKgG0l19Q2MzZOtbeycaF3cvDv9A0oBBYmIgyKq1GLiEpJSufBgYPkXfUgKsCGLWVLQ5hcrtGtUkNw05RWUxO-Qj-CJiFRAAGhA6KwwAC2MAUCAA2iAALIQABuTxcIAAun9pJgoACwfBwQBmABsAAYkZ4gA))
- Vemos que este problema se podría resolver de otra forma con diferente Ux y tener un radio group para una unica seleccion no tiene mucho sentido aparente (con el contexto que tenemos)
- Una forma de resolverlo si queréis mantener la distributiva con elementos disabled podría ser algo así → [Ejemplo](https://mistica-web.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8AlAhlAlhA4gJwgFcAHAAlgDM1CAbAFwDU0bCYBeAHRACM0A7fmi4A%2BDn1KkkAIQgAPGFBQQA7gBlMAZzqjxEyTPmKVYvXrQaNMOm2BIAkpD4BlABYRiKGAHNaaPKQB6YQBfE1NSOkw6GnYuKUEBLjDTWA0wPExiSIg%2BThAATRgaGmMQZL0cgAU8GAtyoPLpOQUlZXKJLA00bhiodtJzS2tbBxzXd08fGj9AkP7I6NiQAEFiYhik3RTa9Mzs3K58GBg%2BTfDSKpq6rdmwpACDFpV1LR179CxcAhIdEAAaEB0FwwAC2tQQAG0QABZCAAN00dD8IAAugDlJgoECNJCAMwANgADCjgkA)
	- Donde la row actúa directamente como botón a la opción que le interese al usuario navegar eliminando esa selección y por tanto el botón de continuar
